jsettlers.logic/src/main/java/jsettlers/logic/map/grid/objects/MapObjectsManager.java

```diff
@@ -400,18 +401,14 @@ public boolean pushMaterial(short x, short y, EMaterialType materialType) {
 		}
 	}
 
-	public ShortPoint2D pushMaterialForced(short x, short y, EMaterialType materialType) {
-		IMapArea mapArea = new HexGridArea(x, y, 0, 200);
-
-		for (ShortPoint2D position : mapArea) {
-			short currX = position.x;
-			short currY = position.y;
+	public ShortPoint2D pushMaterialForced(int x, int y, EMaterialType materialType) {
+		return HexGridArea.iterate(x, y, 0, 200, (currX, currY, radius) -> {
```

Comment:
Have you tested the speed improvement of this?

It may be sufficient to replace `return position;` with `return new ShortPoint2D(currX, currY)` - this allows the JIT to inline the loop object.